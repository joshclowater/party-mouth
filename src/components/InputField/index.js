import React from 'react';
import { Field } from 'redux-form/immutable';
import FieldWrapper from '../FieldWrapper';
import Input from '../Input';

export const RenderField = ({
  input,
  label,
  id,
  meta: { touched, error, warning, submitting },
  ...rest
}) => (
  <FieldWrapper
    inputName={input.name}
    label={label}
    touched={touched}
    error={error}
    warning={warning}
  >
    <Input
      {...input}
      {...rest}
      id={id || input.name}
      disabled={!!submitting}
      hasError={touched && !!error}
    />
  </FieldWrapper>
);

export default function InputField(props) {
  return <Field {...props} component={RenderField} />;
}
