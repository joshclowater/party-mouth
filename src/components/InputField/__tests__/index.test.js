import React from 'react';
import { shallow } from 'enzyme';
import InputField, { RenderField } from '../';

it('InputField renders correctly', () => {
  expect(shallow(<InputField name="mock name" />)).toMatchSnapshot();
});

it('RenderField renders correctly', () => {
  expect(
    shallow(
      <RenderField
        input={{ name: 'mockName', mockInputField: 'mock input field value' }}
        label="mock label"
        id="mock id"
        meta={{
          touched: true,
          error: 'mock error',
          warning: 'mock warning',
          submitting: true
        }}
      />
    )
  ).toMatchSnapshot();
});

describe('id is not passed into RenderField', () => {
  const wrapper = shallow(
    <RenderField
      input={{ name: 'mockName', mockInputField: 'mock input field value' }}
      label="mock label"
      meta={{
        touched: true,
        error: 'mock error',
        warning: 'mock warning',
        submitting: true
      }}
    />
  );
  it('uses input.name as the id', () => {
    expect(wrapper.find('#mockName').length).toBe(1);
  });
});
