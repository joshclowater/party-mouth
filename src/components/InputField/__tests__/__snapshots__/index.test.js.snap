// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<Field
  component={[Function]}
  name="mock name"
/>
`;

exports[`RenderField renders correctly 1`] = `
<FieldWrapper
  error="mock error"
  inputName="mockName"
  label="mock label"
  touched={true}
  warning="mock warning"
>
  <styled.input
    disabled={true}
    hasError={true}
    id="mock id"
    mockInputField="mock input field value"
    name="mockName"
  />
</FieldWrapper>
`;
